datastock/_class1_uniformize.py

```diff
@@ -541,7 +541,8 @@ def get_ref_vector_common(
 
             else:
                 # increments
-                val = np.linspace(b0, b1, int(np.ceil((b1-b0)/ld[imin])))
+                # val = np.linspace(b0, b1, int(np.ceil((b1-b0)/ld[imin])))
+                val = lv[1]
                 key_vector = None
```

Comment:
Not sure I agree,

Here keep in mind that `lv` is the list of time vectors, and `lkeyu` is the list of correspinding time vector names.
We are looking for a common time vector on which all time vectors can be interpolated.
That common time vector must:
* Have the finest resolution (not loose any information)
* Be included which the common limits to all time vectors

Lines 525-526 compute the common limits of time vectors
Lines 534-535 identify the time vector with the best resolution
Lines 537-540 check whether the time vector with the finest resolution is fully included in these limits. If so, it is taken as the common time vector
If not, lines 542-546 create a new time vector within the common limits and with the best resolution (hence key_vector = None).

Why was it causing an issue ?
If the issue was that the number of points was 0 or 1, a simpler possible modification can to do:
```
val = np.linspace(b0, b1, int(np.ceil((b1-b0)/ld[imin])) + 1)     (or +2)
```

This way you are sure that the new vector will always have at least 1(or 2 points)

Let me know if that other solution works

Reply:
Is there any time interpolation process that is made after the inversion step ?
If no, it could be an easy way to re-adjust the shape of the time and profiles arrays with the chosen shape.
If yes, could you tell me where it's supposed to be done ?